keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Locatable.kt

```diff
@@ -26,5 +26,13 @@ interface Locatable<T : RegionSpec> : ResourceSpec {
 
 data class Locations<T : RegionSpec>(
   val accountName: String,
+  /**
+   * If not specified here, this should be derived from [subnet] or use a default VPC name.
+   */
+  val vpcName: String?,
+  /**
+   * If not specified here, this should be derived from a default subnet purpose using [vpcName].
+   */
+  val subnet: String?,
   val regions: Set<T>
 )
```

Comment:
VPC name and subnet (purpose) are now both here. That means

1. every region must use a VPC with the same name.
2. every region must use a subnet with the same purpose.
3. annoyingly, the `subnet` property is present even for `SecurityGroupSpec` where it will always be `null` (or at least ignored).